main.py

```diff
@@ -1,24 +1,42 @@
 from heat_transfer import HeatTransfer
+from bayesian_opt import BayesianOptimization
 
 if __name__ == "__main__":
-    ntime_steps = 20000
+    ntime_steps = 200000
     soil_properties = {
-        "n": 0.6,                     # Porosity
-        "lambda_soil": 1.5,           # Thermal conductivity (W/m-K)
-        "cp_soil": 2000,              # Specific heat capacity (J/kg-K)
-        "rhoS": 1850,                 # Density (kg/m^3)
-        "permeability_soil": 1e-16,   # Permeability (m^2)
+    "n": 0.6,
+    "lambda_soil": 1.5,
+    "cp_soil": 2000,
+    "rhoS": 1850,
+    "permeability_soil": 1e-16,
     }
 
     backfill_properties = {
-        "n_backfill": 0.4,                   # Porosity
-        "lambda_backfill": 1.0,              # Thermal conductivity (W/m-K)
-        "cp_backfill": 800,                  # Specific heat capacity (J/kg-K)
-        "rho_backfill": 1900,                # Density (kg/m^3)
-        "permeability_backfill": 1.0E-11,    # Permeability (m^2)
+    "n_backfill": 0.4,
+    "lambda_backfill": 1.0,
+    "cp_backfill": 800,
+    "rho_backfill": 1900,
+    "permeability_backfill": 1.0E-11,
     }
 
-    cable_offset = 0.05
+    target_offset = 0.05
 
-    heat_transfer_sim = HeatTransfer(soil_properties, backfill_properties, cable_offset, ntime_steps)
-    heat_transfer_sim.run_simulation()
+    # Create an instance of the HeatTransfer class
+    heat_transfer_sim = HeatTransfer(soil_properties, backfill_properties, target_offset, ntime_steps)
+
+    # Get the results_dict from the simulation
+    results_dict = heat_transfer_sim.forward()
+    print('opt_start')
+
+    # Create an instance of BayesianOptimization
+    bayesian_optimization = BayesianOptimization(results_dict, target_offset)
```

Comment:
you can rename this to `opt`
```suggestion
     opt = BayesianOptimization(results_dict, target_offset)
```